### ISSUE TITLE
Flat view name collisions possible

### ISSUE THREAD

**Issue by acspike** [orbeon/orbeon-forms#2439]
Two similarly named forms in the same app can map to the same flat view name if the names are sufficiently long and share a sufficiently long prefix.

**Comment by avernet** [comment]
Good point @acspike, the [flat view](http://doc.orbeon.com/form-runner/persistence/flat-view.html) name mangling scheme makes sure to avoid conflicts with column names, but indeed, conflicts are still possible on the view name. Is this a possible problem you envisioned, or something you've hit in practice.

**Comment by acspike** [comment]
Hit in practice, but it isn't a serious issue. I prefer to make views with more appropriate datatypes by hand, but I usually base those upon the flat view to save typing. With similarly named forms, I simply need to make sure that the view definition I want to use was the last form published.

**Comment by avernet** [comment]
Got it. BTW, could the "more appropriate datatype" be picked automatically, or is that something that really needs to be done manually? And are you saying that you're using that feature with PostgreSQL? We've only tested it on Oracle and DB2; I am am somewhat surprised there; quid?

**Comment by acspike** [comment]
Yes, using the flat views with PostgreSQL. Yes, I think in many cases more appropriate datatypes could be chosen automatically because the fields on forms have associated datatypes. Common changes I make include:
- changing datatypes
- creating additional views for grid and section iterations
- concatenating checkboxes' values together (using multiple checkbox controls to fake a multi-column checkbox layout)
- removing blank fields resulting from explanation controls

**Comment by avernet** [comment]
Agreed, some of those things could be done automatically. BTW, if you'd like to give a shot at improving this, the code is in [`FlatView.scala`](https://github.com/orbeon/orbeon-forms/blob/master/src/main/scala/org/orbeon/oxf/fr/relational/crud/FlatView.scala), and improving it seems pretty realistic.

**Comment by avernet** [comment]
[+1 from community](https://twitter.com/BankJovita)
